app/clients/zendesk_sell.py

```diff
@@ -82,10 +110,113 @@ def send_lead(self, contact: ContactRequest) -> int:
+
+    def delete_contact(self, contact_id: int) -> None:
+
+        if not self.api_url or not self.token:
+            current_app.logger.warning(f'Did not delete contact[{contact_id}] from zendesk')
+            return
+
+        # The API and field definitions are defined here: https://developers.getbase.com/docs/rest/reference/contacts
+        try:
+            response = requests.delete(
+                url=urljoin(self.api_url, f'/v2/contacts/{contact_id}'),
+                headers={'Accept': 'application/json', 'Content-Type': 'application/json'},
+                auth=BearerAuth(token=self.token),
+                timeout=5
+            )
+            response.raise_for_status()
+        except requests.RequestException:
+            current_app.logger.warning(f'Failed to delete zendesk sell contact: {contact_id}')
+
+    def upsert_deal(self, contact_id: int, service: Service, stage_id: int) -> Optional[int]:
+        # The API and field definitions are defined here: https://developers.getbase.com/docs/rest/reference/deals
+        if not self.api_url or not self.token:
+            current_app.logger.warning('Did not upsert deal to zendesk')
+            return None
+
+        try:
+            response = requests.post(
+                url=urljoin(self.api_url, f'/v2/deals/upsert?contact_id={contact_id}&name={escape(service.name)}'),
+                data=json.dumps(ZenDeskSell._generate_deal_data(contact_id, service, stage_id)),
+                headers={'Accept': 'application/json', 'Content-Type': 'application/json'},
+                auth=BearerAuth(token=self.token),
+                timeout=5
+            )
+            response.raise_for_status()
+        except requests.RequestException:
+            current_app.logger.warning('Failed to create zendesk sell deal')
+            return None
+
+        # response validation
+        try:
+            resp_data = json.loads(response.text)
+            assert 'data' in resp_data, 'Missing "data" field in response'
+            assert 'id' in resp_data['data'], 'Missing "id" field in response'
+            assert 'contact_id' in resp_data['data'], 'Missing "contact_id" field in response'
+
+            return resp_data['data']['id']
+
+        except (json.JSONDecodeError, AssertionError):
+            current_app.logger.warning(f'Invalid response: {response.text}')
+            return None
+
+    def send_create_service(self, service: Service, user: User) -> bool:
+        # Upsert a contact (create/update). Only when this is successful does the software upsert a deal
+        # and link the deal to the contact.
+        # If upsert deal fails go back and delete the contact ONLY if it never existed before
+        contact_id, is_created = self.upsert_contact(user)
+        if not contact_id:
+            return False
+
+        # 11826762 is a zendesk number to signify "Created Trial"
+        deal_id = self.upsert_deal(contact_id, service, 11826762)
```

Comment:
Can you add a class constant for the magic number? It'll be more explicit and you could remove a comment.